Microsoft.Toolkit.Uwp.Input.GazeInteraction/IGazeFilter.h

```diff
@@ -0,0 +1,66 @@
+//Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
+//See LICENSE in the project root for license information.
+
+#pragma once
+
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+
+BEGIN_NAMESPACE_GAZE_INPUT
+
+/// <summary>
+/// This struct encapsulates the location and timestamp associated with the user's gaze 
+/// and is used as an input and output parameter for the IGazeFilter::Update method
+/// </summary>
+public ref struct GazeFilterArgs sealed
+{
+    /// <summary>
+    /// The current point in the gaze stream
+    /// </summary>
+    property Point Location {Point get() { return _location; }}
+
+    /// <summary>
+    /// The timestamp associated with the current point
+    /// </summary>
+    property TimeSpan Timestamp {TimeSpan get() { return _timestamp; }}
+
+internal:
+
+    GazeFilterArgs(Point location, TimeSpan timestamp)
+    {
+        _location = location;
+        _timestamp = timestamp;
+    }
+
+private:
+
+    Point _location;
+    TimeSpan _timestamp;
+};
+
+// Every filter must provide an Wpdate method which transforms sample data 
+// and returns filtered output
+public interface class IGazeFilter
+{
+    GazeFilterArgs^ Update(GazeFilterArgs^ args);
+    void LoadSettings(ValueSet^ settings);
+};
+
+
+// Basic filter which performs no input filtering -- easy to
+// use as a default filter.
+public ref class NullFilter sealed : public IGazeFilter
```

Comment:
All the other filters are internal. Should this be too?

Reply:
Yes, Thanks for catching it. I will submit a fix soon.

Reply:
That means IGazeFilter and GazeFilterArgs are not use anywhere else either

Reply:
Fixed and submitted. I also fixed the PointerState enum with explicit values. Thanks for the careful review. 
Please merge if everything else is good. Thanks.